system/loggerd/tests/test_uploader.py

```diff
@@ -89,6 +93,34 @@ def test_upload(self):
 
     self.assertTrue(log_handler.upload_order == exp_order, "Files uploaded in wrong order")
 
+  def test_upload_metered_connection_no_cellular_allowed(self):
+    self.gen_files(lock=False)
+
+    sm = messaging.SubMaster(['deviceState'])
+    sm.update(0)  # timeout=0 during tests
+
+    msg = messaging.new_message('deviceState')
+    # https://github.com/commaai/cereal/blob/d11688a90a1cebacb3fe00dcfbba5f27551b2d89/log.capnp#L308
+    msg.deviceState = sm['deviceState']
+    # Keep everything the same, except `.networkMetered` is now True
+    msg.deviceState.networkMetered = True
```

Comment:
…as it stands I'm a bit stuck as to how one can mock the `deviceState` itself so as to tweet behavior with and without `networkMetered`

(If it comes down to it I could purchase a C3X and test live, but that doesn't improve the test suite itself)